addons/account/controllers/mail.py

```diff
@@ -24,7 +24,7 @@ def _redirect_to_record(cls, model, res_id, access_token=None):
             except AccessError:
                 if record_sudo.access_token and consteq(record_sudo.access_token, access_token):
                     record_action = record_sudo.with_context(
-                        force_website=True).get_access_action(uid)
```

Comment:
What is the reason for this?

Reply:
This is what is done is saas-11.5:
since we already checked if the current user has access or not, we don't need to do it again if there is an AccessError, on the opposite we need to make sure we can display the record if the user is public and has a token

If I didn't do it in sale, it must be a mistake on my part :)

https://github.com/odoo/odoo/blob/saas-11.5/addons/mail/controllers/main.py#L79
https://github.com/odoo/odoo/blob/saas-11.5/addons/portal/controllers/mail.py#L146